web/context.go

```diff
@@ -26,40 +26,40 @@ type requestEvent struct {
 	Extra         map[string]interface{} `json:",omitempty"`
 }
 
-func (e *requestEvent) AddError(msg string, args ...interface{}) {
+func (e *RequestEvent) AddError(msg string, args ...interface{}) {
 	e.Errors = append(e.Errors, fmt.Sprintf(msg, args...))
 }
 
-type wfeHandlerFunc func(context.Context, *requestEvent, http.ResponseWriter, *http.Request)
+type WFEHandlerFunc func(context.Context, *RequestEvent, http.ResponseWriter, *http.Request)
 
-func (f wfeHandlerFunc) ServeHTTP(e *requestEvent, w http.ResponseWriter, r *http.Request) {
+func (f WFEHandlerFunc) ServeHTTP(e *RequestEvent, w http.ResponseWriter, r *http.Request) {
 	ctx := context.TODO()
 	f(ctx, e, w, r)
 }
 
 type wfeHandler interface {
-	ServeHTTP(e *requestEvent, w http.ResponseWriter, r *http.Request)
+	ServeHTTP(e *RequestEvent, w http.ResponseWriter, r *http.Request)
 }
 
-type topHandler struct {
-	wfe wfeHandler
-	log blog.Logger
-	clk clock.Clock
+type TopHandler struct {
```

Comment:
Random question: I don't think this is an issue but are all these fields exported just so that this struct doesn't need a constructor method?

Reply:
Yeah, this felt like the most straightforward way to do the refactor. I'd also be happy to change it to a constructor method if you like.

Reply:
I think a constructor is worthwhile here